Demos/Demo_Pointers/ResourcePointerContext.cpp

```diff
@@ -33,18 +33,23 @@ namespace demo
 				MyGUI::xml::ElementEnumerator item = info->getElementEnumerator();
 				while (item.next("Item"))
 				{
-					mPointers[item->findAttribute("name")] = item->getContent();
+					std::string_view key = item->findAttribute("name");
+					auto it = mPointers.find(key);
+					if (it == mPointers.end())
+						mPointers.emplace(key, item->getContent());
+					else
+						it->second = item->getContent();
```

Comment:
I at one point debated adding a template function to replicate `map[key] = value`, but I'm also curious how many of these instances could be replaced by a simple `emplace` because duplicate keys don't exist or shouldn't exist anyway.

Reply:
Yeah, other places, where you replaced `[]` with single `emplace` are definitely good. The only one I don't like much is when a trivial and easy to read code `map[key] = value` is replaced with 5 lines of some code with ifs and variables

Reply:
And your code does just duplicated operator[] implementation. I understand that initially it was due to string_view -> string extra conversion, but loss of code simplicity definetely doesn't worth it

Reply:
The syntax is definitely a pain, but using this kind of if/else instead of `[]` does actually avoid a useless copy. Somehow the standard managed to give us `insert_or_assign` but didn't add a shorthand for this.

Reply:
> using this kind of if/else instead of [] does actually avoid a useless copy

Yeah, I understand that. But the code quality loss is higher that possible performance gain, unless this is some hot path and it actually almost never is